8Knot/pages/index/index_callbacks.py

```
@@ -734,6 +1128,70 @@ def hide_search_status_when_loaded(_):
+    prevent_initial_call=True,
+)
+def update_search_progress_indicator(search_value):
+    """
+    Update the search progress indicator with adaptive debounce information.
+    This provides visual feedback to users about search state and expected wait time.
+    """
+    if not search_value:
+        return ["", {"display": "none"}]
+
+    # Get adaptive debounce time for this query
+    debounce_ms = get_adaptive_debounce_time(search_value)
```

Ruff (0.14.7)

Local variable `debounce_ms` is assigned to but never used

Remove assignment to unused variable `debounce_ms`

(F841)